### DISCUSSION TITLE
Search functionality on the contents of a file

### DISCUSSION THREAD

**Discussion by pvk2410** [strapi/strapi#7585] [Legacy Topics]
Hello There,

Do we have any support to search for a specific word / phrase from the content of the assets uploaded in media library?
For example : I have uploaded a word document with content " This is for test". I would like to search for this assets from media library via an API , by giving a part of the file content as the search string.

If I try with - http://localhost:1337/uploads?_q=Test ,i get a "Not found" response..

I believe we can search only for a filename but not from the content within the file.

Is this possible at all? 

If not directly, do we have any sample code which can achieve this objective?

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Reply by pvk2410** [reply]
Sure, thank you so much for this correction.

**Comment by derrickmehaffy** [comment]
Currently it is not possible to search the contents of a file no, Strapi doesn't keep any information about the contents only the path to the file and a bit of metadata.
